src/core/src/op/fake_convert.cpp

```diff
@@ -40,6 +40,35 @@ const std::string& FakeConvert::get_destination_type() const {
 void FakeConvert::validate_and_infer_types() {
     OV_OP_SCOPE(v13_FakeConvert_validate_and_infer_types);
     validate_type();
+    for (size_t i = 0; i < get_input_size(); i++) {
+        switch (get_input_element_type(i)) {
+        case element::bf16:
+        case element::f16:
+        case element::f32:
+        case element::dynamic:
+            break;
+        default:
+            OPENVINO_THROW("The element type of the input tensor on index ",
+                           i,
+                           " must be a bf16, f16, f32 or dynamic (got ",
+                           get_input_element_type(i),
+                           ").");
+        }
```

Comment:
Make helper for validation which can be re used in `has_evaluate`. Any of input can have different type or should be same?
There is `merge` function for types.

Reply:
Good catch about `merge`, all inputs should have compatible datatypes. As for helper for `has_evaluate` it is in progress